### PR TITLE
cross-region: only fail the test if a critical error occurs

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve? 

Currently, there are many test failures that are caused by timeouts due to resource constraints. To improve test success rate and focus on the TSO consistency check, we need to handle this situation.

### What is changed and how does it work?

- Only fail the cross-region test if a critical error occurs.
- Wait for leader election without any timeout.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Review by Yisaer — COMMENTED** [review]
I think we can set tikv/tidb replicas to 0 currently.

File: testcase/cross-region/tso.go [hunk 1]
```diff
@@ -352,31 +405,33 @@ func (c *crossRegionClient) waitAllocatorReady(dcLocations []string) error {
 }
 
 func (c *crossRegionClient) waitLeader(name string) error {
-	return util2.WaitUntil(fmt.Sprintf("%s leader election", name), func() bool {
+	for {
```
Comment (Yisaer) [inline testcase/cross-region/tso.go:408]: There will be no timeout

File: testcase/cross-region/tso.go [hunk 2]
```diff
@@ -352,31 +405,33 @@ func (c *crossRegionClient) waitAllocatorReady(dcLocations []string) error {
...
 		}
-		return true
-	})
+		time.Sleep(30 * time.Second)
+	}
 }
 
 func (c *crossRegionClient) waitAllocator(name, dcLocation string) error {
-	return util2.WaitUntil(fmt.Sprintf("%s of %s allocator election", name, dcLocation), func() bool {
+	for {
```
Comment (Yisaer) [inline testcase/cross-region/tso.go:422]: ditto.

**Review by Yisaer — COMMENTED** [review]
Are the replicas of tikv/tidb set to 0?

File: testcase/cross-region/tso.go (hunk 1 above)
Comment (JmPotato) [reply testcase/cross-region/tso.go:408]: I think it's unnecessary since the CI test of TiPocket will kill it automatically when it's timeout.

**Comment by JmPotato** [comment]
@Yisaer PTAL.

**Review by Yisaer — APPROVED** [review]

**Pull request merged**